67_RayQueryGeometry/app_resources/render.comp.hlsl

```diff
@@ -0,0 +1,159 @@
+#include "common.hlsl"
+
+#include "nbl/builtin/hlsl/spirv_intrinsics/core.hlsl"
+
+struct SPushConstants
+{
+    SGeomBDA geom[OT_COUNT];
+};
+
+[[vk::push_constant]] SPushConstants pc;
+
+[[vk::binding(0, 0)]]
+RaytracingAccelerationStructure topLevelAS;
+
+[[vk::binding(1, 0)]]
+cbuffer CameraData
+{
+    SCameraParameters params;
+};
+
+[[vk::binding(2, 0)]] RWTexture2D<float4> outImage;
+
+float3 unpackNormals3x10(uint32_t v)
+{
+    // I have no idea how to unpack this correctly
+    // host side changes float32_t3 to EF_A2B10G10R10_SNORM_PACK32
+    float3 n = (float3)0;
+    const uint mask = 0x3FF;    // 10 bits
+    n.x = float(asint((v >> 20) & mask));
+    n.y = float(asint((v >> 10) & mask));
+    n.z = float(asint((v >> 0) & mask));
+    return normalize(n);
+}
+
+// How the normals are packed
+// cube                     uint32_t - 8, 8, 8, 8
+// sphere, cylinder, arrow  uint32_t - 10, 10, 10, 2
+// rectangle, disk          float3
+// cone                     uint32_t - 10, 10, 10, 2
+// icosphere                float3
+static const uint indexTypes[OT_COUNT]      = { 0, 1, 0, 0, 2, 0, 0, 1 };
+static const uint vertexStrides[OT_COUNT]   = { 24, 28, 28, 32, 32, 28, 20, 32 };
+static const uint byteOffsets[OT_COUNT]     = { 18, 24, 24, 20, 20, 24, 16, 12 };
+
+[numthreads(WorkgroupSize, WorkgroupSize, 1)]
+void main(uint32_t3 threadID : SV_DispatchThreadID)
+{
+    uint2 coords = threadID.xy;
+    uint2 resolution;
+    outImage.GetDimensions(resolution.x, resolution.y);
+    float2 texCoords = float2(float(coords.x) / resolution.x, 1.0 - float(coords.y) / resolution.y);
+
+    if (any(coords >= resolution))
+        return;
+
+    float4 color = float4(0, 0, 0, 1);
+
+    float4 NDC = float4(texCoords * float2(2.0, -2.0) + float2(-1.0, 1.0), 0.0, 1.0);
+    float3 targetPos;
+    {
+        float4 tmp = mul(params.invMVP, NDC);
+        targetPos = tmp.xyz / tmp.w;
+        NDC.z = 1.0;
+    }
+
+    RayDesc ray;
+    ray.TMin = 0.01;
+    ray.TMax = 1000.0;
+    ray.Origin = params.camPos;
+    ray.Direction = normalize(targetPos - params.camPos);
+
+    RayQuery<RAY_FLAG_FORCE_OPAQUE> query;
+    query.TraceRayInline(topLevelAS, 0, 0xFF, ray);
+
+    while (query.Proceed()) {}
+
+    if (query.CommittedStatus() == COMMITTED_TRIANGLE_HIT)
+    {
+        const int instID = query.CommittedInstanceIndex();
+        const int primID = query.CommittedPrimitiveIndex();
```

Comment:
now time to do this in Inline SPIR-V and add the inline SPIR-V intrinsics to our `spirv_intrinsics`